moto/glue/models.py

```diff
@@ -375,8 +406,186 @@ def register_schema_version(self, schema_id, schema_definition):
+                        return get_schema_version_dict
+            raise SchemaVersionNotFoundFromSchemaVersionIdException(schema_version_id)
+
+        # GetSchemaVersion using VersionNumber
+        schema = self.registries[registry_name].schemas[schema_name]
+        for schema_version in schema.schema_versions.values():
+            if (
+                version_number == schema_version.version_number
+                and schema_version.schema_version_status != DELETING_STATUS
+            ):
+                get_schema_version_dict = schema_version.get_schema_version_as_dict()
+                get_schema_version_dict["DataFormat"] = schema.data_format
+                return get_schema_version_dict
+        raise SchemaVersionNotFoundFromSchemaIdException(
+            registry_name, schema_name, schema_arn, version_number, latest_version
+        )
+
+    def get_schema_by_definition(self, schema_id, schema_definition):
+        """GetSchemaByDefinition API"""
+        # Validate SchemaId
+        validate_schema_definition_length(schema_definition)
+        registry_name, schema_name, schema_arn = validate_schema_id(
+            schema_id, self.registries
+        )
+
+        # Get Schema By Definition
+        schema = self.registries[registry_name].schemas[schema_name]
+        for schema_version in schema.schema_versions.values():
+            if (
+                schema_definition == schema_version.schema_definition
+                and schema_version.schema_version_status != DELETING_STATUS
+            ):
+                get_schema_by_definition_dict = (
+                    schema_version.get_schema_by_definition_as_dict()
+                )
+                get_schema_by_definition_dict["DataFormat"] = schema.data_format
+                return get_schema_by_definition_dict
+        raise SchemaNotFoundException(schema_name, registry_name, schema_arn)
+
+    def put_schema_version_metadata(
+        self, schema_id, schema_version_number, schema_version_id, metadata_key_value
+    ):
+        """PutSchemaVersionMetadata API"""
+        # Validate metadata_key_value and schema version params
+        (
+            metadata_key,
+            metadata_value,
+        ) = validate_schema_version_metadata_pattern_and_length(metadata_key_value)
+        (
+            schema_version_id,
+            registry_name,
+            schema_name,
+            schema_arn,
+            version_number,
+            latest_version,
+        ) = validate_schema_version_params(
+            self.registries, schema_id, schema_version_id, schema_version_number
+        )
+
+        # PutSchemaVersionMetadata using SchemaVersionId
+        if schema_version_id:
+            for registry in self.registries.values():
+                for schema in registry.schemas.values():
+                    if schema.schema_versions.get(schema_version_id, None):
+                        validate_number_of_schema_version_metadata_allowed(
```

Comment:
This part is a bit hard to read, with all the newlines.
Maybe we can extract `schema.schema_versions[schema_version_id].metadata` into a temporary variable? That may help a lot in improving the readability

Reply:
Yeah. That really improves the readability.